libindy/src/services/wallet/iterator.rs

```diff
@@ -9,17 +9,17 @@ use super::encryption::{decrypt_tags};
 use utils::crypto::chacha20poly1305_ietf::ChaCha20Poly1305IETF;
 
 
-pub(super) struct WalletIterator<'a> {
-    storage_iterator: Box<StorageIterator + 'a>,
-    keys: &'a Keys
+pub(super) struct WalletIterator {
+    storage_iterator: Box<StorageIterator>,
+    keys: Keys,
 }
 
 
-impl<'a> WalletIterator<'a> {
-    pub fn new(storage_iter: Box<StorageIterator + 'a>, keys: &'a Keys) -> Self {
+impl WalletIterator {
+    pub fn new(storage_iter: Box<StorageIterator>, keys: &Keys) -> Self {
         WalletIterator {
             storage_iterator: storage_iter,
-            keys: keys
+            keys: keys.clone(),
```

Comment:
I know you did it because of lack of time, but I am thinking that maybe keys should be shared pointer (Rc)

Not to avoid unnecessary copying just, but also it might be safer to have keys at one place in memory, not all over the place possibly

Reply:
Exactly. In new one commit the one more same place with `TODO` marker is added. I will also put TODO here.